mysql-test/suite/innodb/t/bug84563.test

```diff
@@ -0,0 +1,35 @@
+--source include/have_innodb.inc
+
+SET GLOBAL innodb_lock_wait_timeout = 1;
+SET GLOBAL innodb_print_lock_wait_timeout_info = true;
+
+CREATE TABLE t (i INT) ENGINE = InnoDB;
+INSERT INTO t (i) VALUES(1);
+
+START TRANSACTION;
+SELECT * FROM t WHERE i = 1 LOCK IN SHARE MODE;
+
+--connect(con0, localhost, root,,)
+START TRANSACTION;
+--send DELETE FROM t WHERE i = 1;
+
+--connection con0
+--error ER_LOCK_WAIT_TIMEOUT
+--reap
+
+--connection default
+DELETE FROM t WHERE i = 1;
+COMMIT;
+
+--connection con0
+COMMIT;
+
+--echo Grepping the error log for lock wait timeout info
+--let SEARCH_FILE= $MYSQLTEST_VARDIR/tmp/bug84563.err
+--let SEARCH_PATTERN= Lock wait timeout info:
+--source include/search_pattern_in_file.inc
+
+DROP TABLE t;
+SET GLOBAL innodb_lock_wait_timeout = default;
```

Comment:
Save these variables into xxx_saved and restore from them rather than from defaults

Reply:
fixed